ibc/client/submodule.go

```diff
@@ -0,0 +1,225 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/pokt-network/pocket/ibc/client/types"
+	"github.com/pokt-network/pocket/ibc/path"
+	core_types "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+var (
+	_                  modules.ClientManager = &clientManager{}
+	allowedClientTypes                       = make(map[string]struct{}, 0)
+)
+
+func init() {
+	allowedClientTypes[types.WasmClientType] = struct{}{}
+}
+
+type clientManager struct {
+	base_modules.IntegrableModule
+
+	logger *modules.Logger
+}
+
+func Create(bus modules.Bus, options ...modules.ClientManagerOption) (modules.ClientManager, error) {
+	return new(clientManager).Create(bus, options...)
+}
+
+// WithLogger sets the logger for the clientManager
+func WithLogger(logger *modules.Logger) modules.ClientManagerOption {
+	return func(m modules.ClientManager) {
+		if mod, ok := m.(*clientManager); ok {
+			mod.logger = logger
+		}
+	}
+}
+
+func (*clientManager) Create(bus modules.Bus, options ...modules.ClientManagerOption) (modules.ClientManager, error) {
+	c := &clientManager{}
+
+	for _, option := range options {
+		option(c)
+	}
+
+	c.logger.Info().Msg("👨 Creating Client Manager 👨")
+
+	bus.RegisterModule(c)
+
+	return c, nil
+}
+
+func (c *clientManager) GetModuleName() string { return modules.ClientManagerModuleName }
+
+// CreateClient creates a new client with the given client state and initial
+// consensus state and initialises it with a unique identifier in the IBC client
+// store and emits an event to the Event Logger
+func (c *clientManager) CreateClient(
+	clientState modules.ClientState, consensusState modules.ConsensusState,
+) (string, error) {
+	// Check if the client type is allowed
+	if !isAllowedClientType(clientState.ClientType()) {
+		return "", fmt.Errorf("client type %s is not supported", clientState.ClientType())
+	}
+
+	// Generate a unique identifier for the client
+	identifier := path.GenerateClientIdentifier()
+
+	// Retrieve the client store prefixed with the client identifier
+	prefixed := path.ApplyPrefix(core_types.CommitmentPrefix(path.KeyClientStorePrefix), identifier)
+	clientStore, err := c.GetBus().GetIBCHost().GetProvableStore(string(prefixed))
+	if err != nil {
+		return "", err
+	}
+
+	// Initialise the client with the clientState provided
+	if err := clientState.Initialise(clientStore, consensusState); err != nil {
+		c.logger.Error().Err(err).Str("identifier", identifier).
+			Msg("failed to initialize client")
```

Comment:
I think we should add `ibc client` in all these places (loggers, errors, shared interfaces, etc...)

I know your mind is in "IBC land", but think of how it was trying to debug P2P issues when you had less context there?